src/src/Helpers/HTTPResponseParser.cpp

```diff
@@ -0,0 +1,397 @@
+#  endif // ifdef FEATURE_USE_DOUBLE_AS_ESPEASY_RULES_FLOAT_TYPE
+
+  // Open the `json.keys` file
+#  ifdef ESP8266
+  const String fileName = F("json.keys");
+#  else // ifdef ESP8266
+  const String fileName = F("/json.keys");
+#  endif // ifdef ESP8266
+
+  if (!fileExists(fileName)) {
+    addLogMove(LOG_LEVEL_ERROR, strformat(F("%s does not exist!"), fileName.c_str()));
+    return;
+  }
+  File keyFile = ESPEASY_FS.open(fileName, "r");
+
+  int keyCount                   = 0;
+  int successfullyProcessedCount = 0; // Counter for successfully processed keys
+  String csvOutput;                   // Collect values as a CSV string
+
+  while (keyFile.available()) {
+    String key = keyFile.readStringUntil('\n');
+    key.trim(); // Remove extra spaces or newlines
+
+    if (key.isEmpty()) {
+      continue;
+    }
+    int keyNumber = 0;
+
+    // Extract the number prefix (before the first dot)
+    if (key.indexOf(':') != -1) {
+      keyNumber = key.substring(0, key.indexOf(':')).toInt();
+      key       = key.substring(key.indexOf(':') + 1);
+    }
+
+    // Only process keys that match the filterNumber (or process all if no match)
+    if ((numJson > 0) && (keyNumber != numJson)) {
+      continue; // Skip this key if it doesn't match the filter number
+    }
+    keyCount++;
+
+    if (keyCount > MAX_KEYS) {
+      addLogMove(LOG_LEVEL_ERROR, strformat(F("Warning: More than %d keys in %s"), MAX_KEYS, fileName.c_str()));
+    }
+
+    // Process the key and navigate the JSON
+    JsonVariant value = *root;
+    size_t start = 0, end;
+
+    while ((end = key.indexOf('.', start)) != (unsigned int)-1) {
+      String part = key.substring(start, end);
+      value = value[part];
+
+      if (value.isNull()) {
+        break; // Key path is invalid
+      }
+      start = end + 1;
+    }
+
+    // Handle the last part of the key
+    if (!value.isNull()) {
+      String lastPart = key.substring(start);
+      value = value[lastPart];
+      successfullyProcessedCount++;
+    }
+
+    // Append the value to the CSV string if it exists
+    if (!value.isNull()) {
+      if (value.is<int>()) {
+        csvOutput += String(value.as<int>());
+      } else if (value.is<float>()) {
+        csvOutput += doubleToString(value.as<double>(), nr_decimals, 1);
+      } else if (value.is<const char *>()) {
+        csvOutput += String(value.as<const char *>());
+      } else if (value.is<JsonArray>()) {
+        // If the value is an array, iterate over its elements
+        JsonArray array        = value.as<JsonArray>();
+        size_t    arraySize    = array.size(); // Get the total number of elements in the array
+        size_t    currentIndex = 0;            // Track the current index
+
+        for (JsonVariant element : array) {
+          if (element.is<int>()) {
+            csvOutput += String(element.as<int>());
+          } else if (element.is<float>()) {
+            csvOutput += doubleToString(element.as<double>(), nr_decimals, 1);
```

Comment:
For boolean arguments it's 'nicer' (for readability) to pass true/false instead of 1/0